### ISSUE TITLE
Questions regarding OTL??

### ISSUE THREAD

**Issue by offlinehacker** [konodyuk/obsidian-typing#4]
I have seen some refactoring you have done, what is OTL format you replaced yaml with?

**Comment by konodyuk** [comment]
OTL stands for Obsidian Typing Language. YAML syntactically restricted me from using some constructions, and overall was not flexible enough for complex configs, thus I decided to introduce a more human-readable notation, which would look cleaner and closer to a programming language.

As a result, I developed both the OTL with island JSX grammar and an obsidian editor view for this file extension.

<details><summary>OTL syntax example in OTL editor</summary>
<img width="870" alt="image" src="https://user-images.githubusercontent.com/20307856/160096749-bc01e235-13cb-4bea-9bcc-7fec32d21c1f.png">
</details>

Although the language is mostly ready for external usage, I still don't have enough time to document it. Here are some features:
- inline js/jsx scripting
- including js/jsx modules from local files
- multiple inheritance
- generic field types
- custom note creation (initialization) scripts
- type actions (=methods), default and pinned actions
